lib/Runtime/Library/BoundFunction.cpp

```diff
@@ -136,16 +136,24 @@ namespace Js
         Var newVarInstance = nullptr;
         if (callInfo.Flags & CallFlags_New)
         {
-          if (JavascriptProxy::Is(targetFunction))
-          {
-            JavascriptProxy* proxy = JavascriptProxy::FromVar(targetFunction);
-            Arguments proxyArgs(CallInfo(CallFlags_New, 1), &targetFunction);
-            args.Values[0] = newVarInstance = proxy->ConstructorTrap(proxyArgs, scriptContext, 0);
-          }
-          else
-          {
-            args.Values[0] = newVarInstance = JavascriptOperators::NewScObjectNoCtor(targetFunction, scriptContext);
-          }
+            if (args.HasNewTarget())
+            {
+                // target has an overriden new target make a new object from the newTarget
+                RecyclableObject* newTarget = RecyclableObject::UnsafeFromVar(args.Values[args.Info.Count]);
```

Comment:
I'd prefer to use `args.GetNewTarget()` here instead of the correct-but-scary-looking indexing operation `args.Values[args.Info.Count]`.

Reply:
I used args.Values[args.Info.Count] because args.GetNewTarget does:

```cc
if (args.HasNewTarget())
{
  return args.Values[args.Info.Count];
}
else
{
  return args.Values[0];
}
```
And I've just done that relevant check in the if above, perhaps a comment to explain it to avoid the double check?

Reply:
Wow, that is pretty scary-looking.  With no other context, intuition would suggest that’s an OoB access.